src/main/java/Admin.java

```diff
@@ -1,15 +1,16 @@
 import java.util.*;
 public class Admin extends User{
-    private int adminId;
+    protected int id;
+
     public Admin(){
```

Comment:
Init id=-1; username="", etc. for empty constructor, or add that to User and super() (which looks like what you were doing but commented out).   Or just leave out the empty constructor all together as it requires the user to update the id manually at a later time which could cause errors.

Reply:
Yeah, I'm game with leaving out the empty constructor.